### ISSUE TITLE
Miter limit should be parameter with default value, not hardcoded.

### ISSUE THREAD

**Issue by ghmart** [godotengine/godot#29886]
**Godot version:**
3.2

**Issue description:**
When using offset_polygon_2d with JOIN_MITER join type it cuts nesessary geometry sometimes:

![ml](https://user-images.githubusercontent.com/7192155/59737480-e22a6d80-926e-11e9-915e-459cd357628f.png)

On the left is what is needed. On the right is what offsetting polygon gives.

Please, add this! Original library allows miter limit to be set.

**Comment by akien-mga** [comment]
I think that's what #29758 does (@Xrayez)

**Comment by ghmart** [comment]
Yes, that's what #29758 does. My apologies. Thought my question on forums were missed and made this issue.

**Comment by Xrayez** [comment]
@ghmart Yeah, no problem, it just makes it more clear what should be done.
